bin/automatic_review.py

```diff
@@ -13,122 +14,145 @@
 
-    yieldq30 = clarity.get_expected_yield_for_sample(sample['sample_id'])
-    if not yieldq30:
-        return None
+            check = None
+            if metric_value is None:
+                check = False
 
-    yieldq30 = int(yieldq30 / 1000000000)
-    expected_yield, coverage = coverage_values[yieldq30]
-    sample_cfg['clean_yield_q30']['value'] = yieldq30
-    sample_cfg['clean_yield_in_gb']['value'] = expected_yield
-    sample_cfg['median_coverage']['value'] = coverage
-    return sample_cfg
+            elif comparison == '>':
+                check = metric_value >= compare_value
 
+            elif comparison == '<':
+                check = metric_value <= compare_value
 
-def get_failing_metrics(metrics, cfg):
-    passfails = {}
+            elif comparison == 'agreeswith':
+                check = metric_value in (self.reviewable_data[compare_value['key']], compare_value['fallback'])
 
-    for metric in cfg:
-        metric_value = query(metrics, metric.split('.'))
-        comparison = cfg[metric]['comparison']
-        compare_value = cfg[metric]['value']
+            passfails[metric] = 'pass' if check else 'fail'
 
-        check = None
-        if metric_value is None:
-            check = False
+        return sorted(k for k, v in passfails.items() if v == 'fail')
 
-        elif comparison == '>':
-            check = metric_value >= compare_value
 
-        elif comparison == '<':
-            check = metric_value <= compare_value
+class LaneReviewer(Reviewer):
+    def __init__(self, aggregated_lane):
+        self.reviewable_data = aggregated_lane
+        self.run_id = aggregated_lane['run_id']
+        self.lane_number = aggregated_lane['lane_number']
 
-        elif comparison == 'agreeswith':
-            check = metric_value in (metrics[compare_value['key']], compare_value['fallback'])
+    @property
+    def cfg(self):
+        return review_thresholds['run']
 
-        result = 'pass' if check else 'fail'
-        passfails[metric] = result
-
-    return sorted(k for k, v in passfails.items() if v == 'fail')
+    @cached_property
+    def _summary(self):
+        failing_metrics = self.get_failing_metrics()
+        if failing_metrics:
+            return {
+                'reviewed': 'fail',
+                ELEMENT_REVIEW_COMMENTS: 'failed due to ' + ', '.join(failing_metrics)
+            }
+        else:
+            return {'reviewed': 'pass'}
+
+    def report(self):
+        s = '%s %s: %s' % (self.run_id, self.lane_number, self._summary)
+        self.info(s)
+        return s
+
+    def push_review(self):
+        rest_communication.patch_entries(
+            'run_elements',
+            payload=self._summary,
+            where={'run_id': self.run_id, 'lane': self.lane_number}
+        )
 
 
 class RunReviewer:
-    def __init__(self, run_name):
-        self.run_name = run_name
-        self.run_elements_by_lane = rest_communication.get_documents(
+    def __init__(self, run_id):
```

Comment:
RunReviewer is now a wrapper for multiple LaneReviewer objects, so it can be called with the same signature as SampleReviewer.